src/app.test.ts

```diff
@@ -434,13 +454,108 @@ test('POST /tilesets/import multiple times using same source file works', async
 
   t.equal(tilesetGetResponse2.statusCode, 200)
 
+  t.notEqual(importId1, importId2, 'new import is created')
+
   cleanup()
 })
 
 /**
- * /styles tests
+ * /imports tests
  */
 
+test('GET /imports/:importId returns import information', async (t) => {
```

Comment:
this seems to be a flaky test that occasionally hangs in CI (see https://github.com/digidem/mapeo-map-server/runs/7032151116?check_suite_focus=true for example). usually if it occurs, it's on Node 14 but the provided example is 16 🤔

Reply:
may have addressed via #34

Reply:
nevermind, still hanging sometimes. based on https://github.com/digidem/mapeo-map-server/runs/7032628641?check_suite_focus=true#step:5:65, looks like 

```
  const getImportInfoResponse = await server.inject({
    method: 'GET',
    url: `/imports/${createdImportId}`,
  })
```

is potentially hanging, which is helpful

Reply:
I think we can use the `finished` field on that table for storing the time of failure. I think it's better to have an enum field for state (rather than booleans `isComplete` and `didError`). How about a field "state" with values `importing`, `complete`, `error`?

Reply:
Maybe also either a field for an error code or an error message? Error code maybe better, makes it more translatable.

Reply:
perfect, seems reasonable to me 👍

Reply:
> Error code maybe better, makes it more translatable.

Can you elaborate on what you mean by "error code" here? Are we talking about a number that we use an enum for or like a standardized error name e.g. `TIMEOUT_ERROR`?

Reply:
I think a standardized error name is more maintainable (like your example `TIMEOUT_ERROR`), so the field in the table is `errorCode: TIMEOUT_ERROR` vs. `errorMessage: The import timed out`.
Maybe to make everything more robust, there should be a sql statement that runs at startup and updates any imports that are not complete as errored? (e.g. if an app crash happened during import)

Reply:
> Maybe to make everything more robust, there should be a sql statement that runs at startup and updates any imports that are not complete as errored? (e.g. if an app crash happened during import)

makes sense to me. curious what to do for the `finished` field in that case though. is it okay to leave that unfilled and if not, what do we set it to? e.g. `started` timestamp or the current timestamp when this sql statement runs

Reply:
to my knowledge, sqlite doesn't have something like that out of the box. we'd have to create a field and update it either manually, or using a trigger https://www.sqlite.org/lang_createtrigger.html